### ISSUE TITLE
Fix Jackson issues

### ISSUE THREAD

**Issue by mariamihai** [mariamihai/udemy-sbm-beer-service#38]
- create no args constructors for event objects
- add object mapper to the MessageConverter bean (telling Spring to inject the managed instance of the object mapper)

**Comment by mariamihai** [comment]
An issue I've encountered was the fact that the BrewBeerEvent couldn't be deserialized. The error I was getting was:
`
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot Deserialize Instance Of 'BrewBeerEvent' Out Of START_ARRAY Token ...
`
Basically OffsetDateTime wasn't deserialized properly.
**This error was encountered after I added the objectMapper to the messageConverter.**


The first step I took was to add an error handler to get more information about the error:
`- added a new class for a simple custom error handler:
`
@Slf4j
@Service
public class MyErrorHandler implements ErrorHandler {

    @Override
    public void handleError(Throwable t) {
        log.error("Error in listener", t);
    }
}
- added DefaultJmsListenerContainerFactory and added the error handler:
`
   @Bean
    public DefaultJmsListenerContainerFactory jmsListenerContainerFactory(ConnectionFactory connectionFactory,
                                                                          MyErrorHandler errorHandler) {
        DefaultJmsListenerContainerFactory factory = new DefaultJmsListenerContainerFactory();
        factory.setConnectionFactory(connectionFactory);
        **factory.setErrorHandler(errorHandler);**

        return factory;
    }
`
The example I used can be found [here](https://stackoverflow.com/questions/8922532/execution-of-jms-message-listener-failed-and-no-errorhandler-has-been-set).

By adding the error handler, I got more information as the listener was getting a String (as the message) from the queue that couldn't be deserialized to a BrewBeerEvent object. One way to maybe fix this is for the method to have a String parameter that is manually deserialized to a BrewBeerEvent object. 
`
public class BrewBeerListener {

    private final BeerRepository beerRepository;
    private final JmsTemplate jmsTemplate;

    private final ObjectMapper objectMapper;

    @Transactional
    @JmsListener(destination = JmsConfig.BREWING_REQUEST_QUEUE)
    public void listen(BrewBeerEvent brewBeerEvent) {
        BrewBeerEvent brewBeerEvent = null;
        try {
            brewBeerEvent = objectMapper.readValue(string, BrewBeerEvent.class);
            //...
       } catch (IOException e) {
            e.printStackTrace();
       }
    }
`
I actually expected this to fail and get even more information about the issue but the object was deserialized properly.


Currently, there are no issues when serializing or deserializing the messages and I am not using the error handler or deserializing manually the String (the returned TextMessage by jms).
Not sure if this is an injection issue or maybe a build issue.
